### DISCUSSION TITLE
How to blend 2 different materials?

### DISCUSSION THREAD

**Discussion by maudzung** [google/filament#5865] [Q&A]
Hi authors,

I can render 2 objects separately. Each object has its image textures and its mask (weighted from 0 to 1).
In the .mat file, I define as below:
```
material {
    name : BlendingSample,
    parameters : [
        {
            type : sampler2d,
            name : albedo
        },
        {
            type : sampler2d,
            name : mask
        }
    ],
    requires : [
        uv0
    ],
    shadingModel : unlit
}

fragment {
    void material(inout MaterialInputs material) {
        prepareMaterial(material);
        float weight = texture(materialParams_mask, getUV0()).r;
        material.baseColor.rgb = texture(materialParams_albedo, getUV0()).rgb * weight;
    }
}
```
Can you please guide me on how to blend them with their weights?
Thank you so much!

**Comment by bejado** [comment]
You'll want to review the Blending and Transparency sections of the Filament material guide, in particular the [`blending`](https://google.github.io/filament/Materials.html#materialdefinitions/materialblock/blendingandtransparency:blending) and [`transparent`](https://google.github.io/filament/Materials.html#materialdefinitions/materialblock/blendingandtransparency:transparency) properties.
